include/aws/common/atomics_msvc.inl

```diff
@@ -65,8 +65,8 @@
  * this use case.
  */
 
-#pragma warning(push)
-#pragma warning(disable : 4201) /* nameless union */
+#include <Windows.h>
```

Comment:
I think we would do well to just forward declare the WINAPI functions required, instead of pulling in Windows.h here, naked, and foisting it on users. They are very stable, and should be consistent across windows API revs.

Reply:
Turns out we don't need to! If we just prefix each function with `_`, it uses the compiler intrinsic and we don't need to include Windows.h OR forward declare. Because all Windows.h is:
```c
#define InterlockedExchangePointer _InterlockedExchangePointer
/* etc. */
```